### PR TITLE
WSTEAM1-676: Add fallback for default image

### PR DESCRIPTION
**Pull request by shayneahchoon**
Resolves JIRA [WSTEAM1-676](https://jira.dev.bbc.co.uk/browse/WSTEAM1-676)

Overall changes
======
Add a fallback image for frosted promos: 
Screenshot 2023-11-01 at 16 39 58

Code changes
======
- _Adds optional chaining to `OptimoPromoFormatter` in `src/app/components/FrostedGlassPromo/withData.tsx`._
- _Adds default image props to `buildImageProperties` in `src/app/components/FrostedGlassPromo/withData.tsx`._

Testing
======
1. Turn on faulty 'informate' section for `https://www.test.bbc.com/pidgin/articles/cv6yjn1z042o`.
2. Go to `https://www.test.bbc.com/pidgin/articles/cv6yjn1z042o`.
3. Make sure informate section is rendering correctly with 'video 2' showing the default BBC logo. 

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

**Review by amoore108 — APPROVED** [review]

**Review by Isabella-Mitchell — APPROVED** [review]
Tested locally by adding broken features to article fixture data. Looks good!

File: src/app/components/FrostedGlassPromo/index.test.tsx [hunk 1]
```diff
@@ -118,6 +124,32 @@ describe('Frosted Glass Promo', () => {
...
+          optimoPromoFixture.item.locators.canonicalUrl,
+        )}"]`,
+      ),
+    ).toBeInTheDocument();
+    expect(
+      container.querySelector(
+        `img[src="https://ichef.bbci.co.uk/news/400/cpsprodpb/36D1/production/_127933041__63970643_bbc-news-world-service-logo-nc.png.webp"]`,
+      ),
+    ).toBeInTheDocument();
+  });
```
Comment (andrewscfc) [inline src/app/components/FrostedGlassPromo/index.test.tsx:171]: Just regarding this point from the ticket:

> Regardless of any behaviour stated above we need to engineer the fix to ensure that where the data is not in an expected format we omit the promo from the section it is included within (e.g. features and analysis); the current situation where it causes the whole page not to render is needs to be avoided in future.

Could we add a test for the scenario where the component returned is completely invalid? So deliberately leave out key information like the headline, url etc that we can't render a promo without.

What we are after is that whatever information is missing we don't throw an error but instead return `null` so we can render with the promo missing.

File: src/app/components/FrostedGlassPromo/withData.tsx [hunk 2]
```diff
@@ -18,8 +18,20 @@ import styles from './styles';
...
+      ? imageProps
+      : {
+          height: null,
+          width: 400,
+          altText: '',
+          path: '/cpsprodpb/36D1/production/_127933041__63970643_bbc-news-world-service-logo-nc.png',
+          locator: null,
+          originCode: null,
+          copyright: null,
+        };
```
Comment (andrewscfc) [inline src/app/components/FrostedGlassPromo/withData.tsx]: Is it worth storing this image in it's own variable so the intention is a little clear so: 
```javascript
const defaultImage = {
          height: null,
          width: 400,
          altText: '',
          path: '/cpsprodpb/36D1/production/_127933041__63970643_bbc-news-world-service-logo-nc.png',
          locator: null,
          originCode: null,
          copyright: null,
        };
```

**Review by andrewscfc — COMMENTED** [review]
Few comments, just applying some defensive practices, nice work!

File: src/app/components/FrostedGlassPromo/index.test.tsx (hunk 1 above)
Comment (shayneahchoon) [reply src/app/components/FrostedGlassPromo/index.test.tsx:171]: Added test.

**Pull request merged**